diff -Npur a/kernel/nvidia/nv-frontend.c b/kernel/nvidia/nv-frontend.c
--- a/kernel/nvidia/nv-frontend.c	2017-07-19 22:43:50.000000000 -0400
+++ b/kernel/nvidia/nv-frontend.c	2017-09-13 10:00:07.552395161 -0400
@@ -15,7 +15,7 @@
 #include "nv-frontend.h"
 
 #if defined(MODULE_LICENSE)
-MODULE_LICENSE("NVIDIA");
+MODULE_LICENSE("GPL");
 #endif
 #if defined(MODULE_INFO)
 MODULE_INFO(supported, "external");
diff -Npur a/kernel/nvidia/nv.c b/kernel/nvidia/nv.c
--- a/kernel/nvidia/nv.c	2017-07-19 22:43:50.000000000 -0400
+++ b/kernel/nvidia/nv.c	2017-09-13 09:59:32.984665063 -0400
@@ -38,7 +38,7 @@
 
 #if (NV_BUILD_MODULE_INSTANCES != 0)
 #if defined(MODULE_LICENSE)
-MODULE_LICENSE("NVIDIA");
+MODULE_LICENSE("GPL");
 #endif
 #if defined(MODULE_INFO)
 MODULE_INFO(supported, "external");
diff -Npur a/kernel/nvidia-drm/nvidia-drm-crtc.c b/kernel/nvidia-drm/nvidia-drm-crtc.c
--- a/kernel/nvidia-drm/nvidia-drm-crtc.c	2017-07-19 22:33:42.000000000 -0400
+++ b/kernel/nvidia-drm/nvidia-drm-crtc.c	2017-09-13 10:02:49.920728781 -0400
@@ -170,7 +170,7 @@ static void nvidia_crtc_enable(struct dr
 static const struct drm_crtc_helper_funcs nv_crtc_helper_funcs = {
     .prepare    = nvidia_crtc_prepare,
     .commit     = nvidia_crtc_commit,
-    .enable     = nvidia_crtc_enable,
+    //.enable     = nvidia_crtc_enable,
     .disable    = nvidia_crtc_disable,
     .mode_fixup = nvidia_crtc_mode_fixup,
 };
@@ -223,10 +223,11 @@ static struct drm_plane *nvidia_plane_cr
         dev,
         plane, crtc_mask, funcs,
         formats, formats_count,
+        NULL,
         plane_type
-#if defined(NV_DRM_INIT_FUNCTIONS_HAVE_NAME_ARG)
+//#if defined(NV_DRM_INIT_FUNCTIONS_HAVE_NAME_ARG)
         , NULL
-#endif
+//#endif
         );
 
     if (ret != 0)
@@ -354,9 +355,9 @@ struct drm_crtc *nvidia_drm_add_crtc(str
                                     &nv_crtc->base,
                                     primary_plane, cursor_plane,
                                     &nv_crtc_funcs
-#if defined(NV_DRM_INIT_FUNCTIONS_HAVE_NAME_ARG)
+//#if defined(NV_DRM_INIT_FUNCTIONS_HAVE_NAME_ARG)
                                     , NULL
-#endif
+//#endif
                                     );
 
     if (ret != 0)
diff -Npur a/kernel/nvidia-drm/nvidia-drm-encoder.c b/kernel/nvidia-drm/nvidia-drm-encoder.c
--- a/kernel/nvidia-drm/nvidia-drm-encoder.c	2017-07-19 22:33:42.000000000 -0400
+++ b/kernel/nvidia-drm/nvidia-drm-encoder.c	2017-09-13 10:03:11.871405426 -0400
@@ -144,9 +144,9 @@ nvidia_encoder_new(struct drm_device *de
     ret = drm_encoder_init(dev,
                            &nv_encoder->base, &nv_encoder_funcs,
                            nvkms_connector_signal_to_drm_encoder_signal(format)
-#if defined(NV_DRM_INIT_FUNCTIONS_HAVE_NAME_ARG)
+//#if defined(NV_DRM_INIT_FUNCTIONS_HAVE_NAME_ARG)
                            , NULL
-#endif
+//#endif
                            );
 
     if (ret != 0)
diff -Npur a/kernel/nvidia-drm/nvidia-drm-linux.c b/kernel/nvidia-drm/nvidia-drm-linux.c
--- a/kernel/nvidia-drm/nvidia-drm-linux.c	2017-07-19 22:33:42.000000000 -0400
+++ b/kernel/nvidia-drm/nvidia-drm-linux.c	2017-09-13 10:03:59.502874219 -0400
@@ -196,7 +196,7 @@ module_init(nv_linux_drm_init);
 module_exit(nv_linux_drm_exit);
 
 #if defined(MODULE_LICENSE)
-  MODULE_LICENSE("MIT");
+  MODULE_LICENSE("GPL");
 #endif
 #if defined(MODULE_INFO)
   MODULE_INFO(supported, "external");
diff -Npur a/kernel/nvidia-modeset/nvidia-modeset-linux.c b/kernel/nvidia-modeset/nvidia-modeset-linux.c
--- a/kernel/nvidia-modeset/nvidia-modeset-linux.c	2017-07-19 22:34:33.000000000 -0400
+++ b/kernel/nvidia-modeset/nvidia-modeset-linux.c	2017-09-13 10:04:30.883842290 -0400
@@ -1285,7 +1285,7 @@ module_init(nvkms_init);
 module_exit(nvkms_exit);
 
 #if defined(MODULE_LICENSE)
-  MODULE_LICENSE("NVIDIA");
+  MODULE_LICENSE("GPL");
 #endif
 #if defined(MODULE_INFO)
   MODULE_INFO(supported, "external");
diff -Npur a/kernel/nvidia-uvm/uvm_common.c b/kernel/nvidia-uvm/uvm_common.c
--- a/kernel/nvidia-uvm/uvm_common.c	2017-07-19 22:43:45.000000000 -0400
+++ b/kernel/nvidia-uvm/uvm_common.c	2017-09-13 10:05:54.699762753 -0400
@@ -388,5 +388,5 @@ module_param(uvm_enable_builtin_tests, i
 MODULE_PARM_DESC(uvm_enable_builtin_tests,
                  "Enable the UVM built-in tests. (This is a security risk)");
 
-MODULE_LICENSE("MIT");
+MODULE_LICENSE("GPL");
 MODULE_INFO(supported, "external");
diff -Npur a/kernel/nvidia-uvm/uvm_unsupported.c b/kernel/nvidia-uvm/uvm_unsupported.c
--- a/kernel/nvidia-uvm/uvm_unsupported.c	2017-07-19 22:43:45.000000000 -0400
+++ b/kernel/nvidia-uvm/uvm_unsupported.c	2017-09-13 10:05:09.588370040 -0400
@@ -171,6 +171,6 @@ static void __exit uvm_unsupported_exit(
 module_init(uvm_unsupported_module_init);
 module_exit(uvm_unsupported_exit);
 
-MODULE_LICENSE("MIT");
+MODULE_LICENSE("GPL");
 MODULE_INFO(supported, "external");
 
